### PR TITLE
fix(server): guard null historyHash in wavelet deserialization

### PR DESCRIPTION
**Pull request by vega113**
## Summary

- **`CoreWaveletOperationSerializer.deserialize(ProtocolHashedVersion)`**: Added null/empty guard — when GWT's JS protobuf runtime returns `null` for an empty `history_hash` field, the code now falls back to `new byte[0]` instead of NullPointerException → `checkState` crash.
- **`Mongo4DeltaStoreUtil.deserializeHashedVersion(Document)`**: When the MongoDB document is missing the `historyhash` field (e.g. older records), `historyHash` was left `null` and passed to `HashedVersion.of()` which rejects it. Now defaults to `new byte[0]`.

## Root cause

GWT stack trace `Token: 1775379766698 null … checkState → $create_3 → deserialize_3 → onWaveletUpdate` traced to `HashedVersion(long, byte[])` rejecting a null `historyHash`. Two call sites could supply null: the protobuf deserializer (GWT JS runtime returns null for empty ByteString) and the Mongo store util (missing document field falls through both instanceof checks).

## Test plan

- [x] New `Mongo4DeltaStoreUtilTest.testDeserializeHashedVersionMissingHashFieldReturnsEmptyHash` — was RED before fix, GREEN after
- [x] New `Mongo4DeltaStoreUtilTest.testDeserializeHashedVersionWithBinaryHashRoundTrips` — happy path unchanged
- [x] New `WaveletOperationSerializerTest.testDeserializeHashedVersionWithEmptyProtobufHash` — contract test for the empty ByteString path
- [x] All 22 tests in `org.waveprotocol.box.server.common.*` pass
- [x] `sbt wave/compile` clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved resilience in data processing by implementing defensive error handling for scenarios where data fields are missing or empty, preventing unexpected failures during serialization operations

* **Tests**
  * Added test coverage for edge cases in data serialization, validating that the system gracefully handles missing and incomplete data without errors

### CODE REVIEW

File: docs/superpowers/specs/2026-04-05-rtl-support-design.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# RTL Language Support Design
+
+**Date:** 2026-04-05  
+**Status:** Approved  
+**Scope:** Right-to-left language rendering (Hebrew, Arabic, etc.) with auto-detection and manual override
+
```
[the remark this answers is not in the document]
Comment (vega113) [reply docs/superpowers/specs/2026-04-05-rtl-support-design.md]: These RTL docs were incidentally on the branch as spec artifacts. After merging main, these files are no longer part of the PR diff — the docs now match main exactly. No action needed.
